### PR TITLE
add tests for LibreSSL

### PR DESCRIPTION
**Pull request by fboundp**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -27,21 +37,39 @@ env:
...
+    - LISP=ecl OPENSSL=libressl-2.6.5 BITS=64
+    - LISP=ecl OPENSSL=libressl-2.8.3 BITS=64
+    - LISP=ecl OPENSSL=libressl-3.0.1 BITS=64
+    - LISP=ecl OPENSSL=libressl-2.2.7 BITS=64
     - LISP=abcl OPENSSL=openssl-0.9.8zh BITS=64
     - LISP=abcl OPENSSL=openssl-1.0.0s BITS=64
     - LISP=abcl OPENSSL=openssl-1.0.2q BITS=64
     - LISP=abcl OPENSSL=openssl-1.1.0j BITS=64
     - LISP=abcl OPENSSL=openssl-1.1.1a BITS=64
+    - LISP=abcl OPENSSL=libressl-2.5.5 BITS=64
```
Comment (avodonosov) [inline .travis.yml]: Why no 2.2.7 for abcl, cmucl and allegro?

File: test/run-on-many-lisps-and-openssls/openssl-releases/build.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ cd ../src
 VERSIONS="$1"
 if [ -z "$VERSIONS" ]
 then
-  VERSIONS="openssl-0.9.8zh openssl-1.0.0s openssl-1.0.2q openssl-1.1.0j openssl-1.1.1a"
+  VERSIONS="openssl-0.9.8zh openssl-1.0.0s openssl-1.0.2q openssl-1.1.0j openssl-1.1.1a libressl-2.2.7 libressl-3.0.1"
 fi
```
Comment (avodonosov) [inline test/run-on-many-lisps-and-openssls/openssl-releases/build.sh]: Why only two libressl versions by default?

File: .travis.yml (hunk 1 above)
Comment (fboundp) [reply .travis.yml]: Copy and paste fail. Fixed.

File: test/run-on-many-lisps-and-openssls/openssl-releases/build.sh (hunk 2 above)
Comment (fboundp) [reply test/run-on-many-lisps-and-openssls/openssl-releases/build.sh]: Simply failed to add them after having tested all versions thru Travis. Fixed.

**Comment by avodonosov** [comment]
@fboundp, merged, thank you

**Pull request merged**